ui/pages/confirmations/components/confirm/info/hooks/use-eip-1559-fees.ts

```diff
@@ -0,0 +1,24 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useState } from 'react';
+import { hexToDecimal } from '../../../../../../../shared/modules/conversion.utils';
+
+export const useEIP1559TxFees = (currentConfirmation: TransactionMeta) => {
+  const [maxFeePerGas, setMaxFeePerGas] = useState(0);
+  const [maxPriorityFeePerGas, setMaxPriorityFeePerGas] = useState(0);
+
+  useEffect(() => {
+    const newMaxFeePerGas = currentConfirmation?.txParams?.maxFeePerGas
+      ? Number(hexToDecimal(currentConfirmation.txParams.maxFeePerGas))
+      : 0;
+
+    const newMaxPriorityFeePerGas = currentConfirmation?.txParams
+      ?.maxPriorityFeePerGas
+      ? Number(hexToDecimal(currentConfirmation.txParams.maxPriorityFeePerGas))
+      : 0;
+
+    setMaxFeePerGas(newMaxFeePerGas);
+    setMaxPriorityFeePerGas(newMaxPriorityFeePerGas);
+  }, [currentConfirmation]);
```

Comment:
We're re-parsing whenever the confirmation object reference changes, but we actually only care about the specific params themselves, so could we be more explicit and more optimised if we change the dependencies to them instead?

```
    const hexMaxFeePerGas = currentConfirmation?.txParams?.maxFeePerGas;
    const hexMaxPriorityFeePerGas = currentConfirmation?.txParams?.maxPriorityFeePerGas;

    return useMemo(() => { ... }, [hexMaxFeePerGas, hexMaxPriorityFeePerGas]);
```